lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/BulkPartitionTask.scala

```diff
@@ -68,7 +68,6 @@ class BulkPartitionTask extends BasePartitionTask {
       val datasetInner: LightGBMDataset = ac.generateDataset(ctx, referenceDataset)
       ctx.trainingCtx.columnParams.groupColumn.foreach(_ => datasetInner.addGroupColumn(ac.getGroups))
```

Comment:
This line looks very suspicious. Since you do `foreach(_ =>...)` the code does not really care about `ctx.trainingCtx.columnParams.groupColumn` (it's not used inside the `foreach` block). In other words, it's just `if (!ctx.trainingCtx.columnParams.groupColumn.isEmpty)` (is not empty).

What's the type of `ctx.trainingCtx.columnParams.groupColumn`? Why is this important to execute `datasetInner.addGroupColumn(ac.getGroups)`?

Reply:
groupColumn is Option[string].  I agree it looks weird to me, but the foreach is basically being used as "if groupColumn has a value do this". 

Is there a better pattern for this that is more readable?  (although loath to change old code that will be deprecated)

Reply:
I don't know any better way than what I proposed 😉 The issue is that `foreach(_ => ...)` does not say you're doing the existence check (something that'd be more clear with good ol' `if`). Hence the comment.

I'm OK if you don't want to change old code 👍